silence/silence.go

```diff
@@ -755,6 +793,9 @@ func (s *Silences) QueryOne(params ...QueryParam) (*pb.Silence, error) {
 // Query for silences based on the given query parameters. It returns the
 // resulting silences and the state version the result is based on.
 func (s *Silences) Query(params ...QueryParam) ([]*pb.Silence, int, error) {
+	s.mtx.Lock()
```

Comment:
Moved the mutex from `query` to `Query` so we can call `query` from `Set` while the mutex is held.